src/parse.rs

```diff
@@ -52,41 +51,57 @@ impl Line {
-            "ARRAY" => Some(format!(
-                "Array[{:#x}][len={}]",
-                object.address, self.length?
-            )),
-            "HASH" => Some(format!("Hash[{:#x}][size={}]", object.address, self.size?)),
-            "STRING" => self.value.map(|v| {
-                let prefix = v
-                    .chars()
-                    .take(40)
-                    .flat_map(|c| {
-                        // Hacky escape to prevent dot format from breaking
-                        if c.is_control() {
-                            None
-                        } else if c == '\\' {
-                            Some('﹨')
-                        } else {
-                            Some(c)
-                        }
-                    })
-                    .collect::<String>();
-                let ellipsis = if v.chars().nth(41).is_some() {
-                    "…"
-                } else {
-                    ""
-                };
-                format!("String[{:#x}][{}{}]", object.address, prefix, ellipsis)
-            }),
-            _ => None,
-        };
-
+        if !trimmed {
+            object.label = match object.kind.as_str() {
+                "CLASS" | "MODULE" | "ICLASS" =>  self
+                    .name
+                    .clone()
+                    .map(|n| format!("{}[{:#x}][{}]", n, object.address, object.kind)),
+                "ARRAY" => Some(format!(
+                    "Array[{:#x}][len={}]",
+                    object.address, self.length?
+                )),
+                "HASH" =>  Some(format!("Hash[{:#x}][size={}]", object.address, self.size?)),
+                "STRING" =>  self.value.as_ref().map(|v| {
+                    let prefix = v
+                        .chars()
+                        .take(40)
+                        .flat_map(|c| {
+                            // Hacky escape to prevent dot format from breaking
+                            if c.is_control() {
+                                None
+                            } else if c == '\\' {
+                                Some('﹨')
+                            } else {
+                                Some(c)
+                            }
+                        })
+                        .collect::<String>();
+                    let ellipsis = if v.chars().nth(41).is_some() {
+                        "…"
+                    } else {
+                        ""
+                    };
+                    format!("String[{:#x}][{}{}]", object.address, prefix, ellipsis)
+                }),
+                _ => None 
+            }
+        }
+        else {
+           object.label= match object.kind.as_str() {
+                "CLASS" | "MODULE" | "ICLASS" => self
+                    .name
+                    .clone()
+                    .map(|n| format!("{}[{}]", n,  object.kind)),
+                "ARRAY" => Some(format!(
+                    "Array[{:#x}][len={}]",
```

Comment:
It looks like we keep the array address here - I assume that's accidental?

Also, it feels a little arbitrary to me to keep the size of arrays and hashes, while losing not just addresses but also the contents of strings. Does it make sense to drop the sizes, or did you have a specific use case for them?

If we do drop them, maybe rather than calling the option `trim`, which doesn't seem super transparent to me, we could call it something like `class-name-only`.

Reply:
Yeah this is my bad. I will fix it. `class-name-only` is better 👌.